samples/Ralltiir/Ralltiir.Server/ClientApp/src/app/components/authorize.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { Button } from 'react-bootstrap';
+import { useDispatch, useSelector } from 'react-redux';
+import { Redirect } from 'react-router-dom';
+import _ from 'underscore';
+
+import { AppState } from '../redux/rootReducer';
+import { createAuthorizeAction, createLogoutAction } from '../redux/user/userActions';
+
+function Authorize() {
+  const dispatch = useDispatch();
+
+  const usersState = useSelector((state: AppState) => state.user);
+  const isLoading = usersState.get("isLoading");
+  const userManager = usersState.get("userManager");
+  const scopes = userManager.settings.scope;
+  const clientId = userManager.settings.client_id;
+
+  const accept = () => {
+    dispatch(createAuthorizeAction(true));
+  };
+  const deny = () => {
+    dispatch(createLogoutAction());
+  };
+  return _.every(usersState.get("loginError"), e => e.error !== "consent_required") ? (
+    <Redirect to="/" />
+  ) : (
+    <div className="authorize">
+      <h2>Authorization</h2>
+      <br />
+      <p className="lead text-left">
+        Do you want to grant <strong>{clientId}</strong> access to your data?
```

Comment:
Hahaha, you can't do that. You can't just use the "client" config of the frontend: these values need to be resolved from the request (e.g by flowing some state between the server and the frontend), otherwise you'll get completely wrong values if someone adds more OIDC clients.

Reply:
That makes sense. Would using the `azp` field from the profile response be alright?

Reply:
Unfortunately no, the data must come from the OIDC request (ideally, flown by the server as I mentioned elsewhere).

Reply:
I'm a little confused... since this is the client, do you mean the OIDC response?

Reply:
Actually no, at this point, we haven't returned an OIDC response to the client yet as the user needs to approve the authorization grant (it's the whole point of this consent page).

What makes things conceptually harder is that your frontend both acts as a regular OIDC client and as the UI for the server. In this specific case, the fact the frontend is also an OIDC client doesn't matter.

See my related remark at https://github.com/openiddict/openiddict-samples/pull/130#discussion_r619846993.